bin/runtime-common/src/messages_xcm_extension.rs

```diff
@@ -28,13 +28,42 @@ use bp_messages::{
 };
 use bp_runtime::{messages::MessageDispatchResult, AccountIdOf, Chain};
 use codec::{Decode, Encode};
-use frame_support::{dispatch::Weight, traits::Get, CloneNoBound, EqNoBound, PartialEqNoBound};
+use frame_support::{
+	dispatch::Weight, traits::Get, weights::RuntimeDbWeight, CloneNoBound, EqNoBound,
+	PartialEqNoBound,
+};
 use scale_info::TypeInfo;
+use sp_std::marker::PhantomData;
 use xcm_builder::{DispatchBlob, DispatchBlobError, HaulBlob, HaulBlobError};
 
 /// Plain "XCM" payload, which we transfer through bridge
 pub type XcmAsPlainPayload = sp_std::prelude::Vec<u8>;
 
+// TODO: below are just rough estimations. Other things also happen there (including hashing and so
```

Comment:
@bkontur I've copypasted this `XcmRouterWeigher` from RBH/WBH code. I think we'll be using it for all bridge hubs, right? Could you please check my big comment below - am I right there?

Right now the biggest problem here is that it only returns the `ref_time` component and the `proof_size` is set to zero. Which is obviously wrong. I see two options to fix it:
- come up with some good estimation of the `proof_size` and return it here. Pros: easy and fast. Cons: inaccurate + we need some mechanism to verify that it is still actual (if something changes in queue implementation);
- remove this `XcmRouterWeigher` at all and add some benchmarks to the runtimes, which will measure the actual weight of the `queue.push()` (for us it'll be the `send_fragment` in the `cumulus-pallet-xcmp-queue`).

WDYT?

Reply:
Submitted https://github.com/paritytech/parity-bridges-common/issues/1986 to track this